dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -104,6 +104,29 @@ export default function Catalog() {
 
   const csvs = operators.csvs;
 
+  // Only one search filter can be set
+  const searchFilter = filters[filterNames.SEARCH]?.toString().replace(tmpStrRegex, ",") || "";
+  const reposFilter = filters[filterNames.REPO]?.join(",") || "";
+
+  // Detect changes in cluster/ns/repos/search and reset the current package
+  // list Note: useEffect is called on every render - the initial render and any
+  // re-renders due to updates. Additionally, any cleanup function returned by
+  // an effect is run prior to re-running the effect as well as on unmount.
+  // https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1
+  // Therefore the reset effect below only has a cleanup function so that we a)
+  // don't reset when the component is initially rendered, and b) do reset
+  // whenever the effect is re-triggered or the component unmounted.
+  useEffect(() => {
+    // Ensure that when this component is unmounted, we remove any catalog state
+    // so that it is reloaded cleanly for the given inputs next time it is
+    // loaded.
+    return function cleanup() {
```

Comment:
Do you know if the cleanup function has necessarily to be `cleanup()` ? In other places we just have `return () => {//whatever};` ?
I mean, sth like: 
```suggestion
    return () => {
```

Reply:
It doesn't have to be a named function, it works fine as an anonymous function, but as in the React docs, I prefer a named one here just so it's obvious to anyone reading the code what the purpose of the returned function is.